### DISCUSSION TITLE
refresh items in a drawer

### DISCUSSION THREAD

**Discussion by TeeFiX42** [zauberzeug/nicegui#3856] [Q&A]
### Question

Hi,
I'm trying to refresh the content of a left_drawer. I checked multiple examples on using the ui.refreshable decorator. But cant find what I'm looking for.
minimum code:
```python
from nicegui import ui
import datetime


class MyUi():
    def __init__(self) -> None:
        self.left_menu = ui.left_drawer().props('bordered')

    @ui.refreshable
    def _build_left_menu(self, text="something"):
        with self.left_menu:
            ui.label(text)      

jjj = MyUi()
jjj._build_left_menu()

ui.button(on_click=lambda: jjj._build_left_menu.refresh(text=datetime.datetime.now()))

ui.run()
```
I just want a label in the left_drawer to be updated in the above example.
But every time I click on the button, a new label is added in the self.left_menu.

Expected behavior:
I would expect that the previous created label in self.left_menu is deleted and a a new label is created when clicking on the button. Why is that not happening?

It works when I put the label in a ui.column() instead of a drawer. I don't understand why it behaves like that. 

How can I update items in a left_drawer object?

**Comment by falkoschindler** [comment]
Hi @TeeFiX42,

Please have a look at https://github.com/zauberzeug/nicegui/issues/3760#issuecomment-2371711304, where I'm explaining the problem with switching the UI context via `with` inside a refreshable container.

**Reply by TeeFiX42** [reply]
Tnx for the tip! I was able to solve my issue.
